torch/distributed/distributed_c10d.py

```diff
@@ -1533,7 +1533,7 @@ def init_process_group(
             backend,
             store,
             group_name,
-            pg_options=pg_options,
+            backend_options=pg_options,
```

Comment:
Since a process group can have multiple backends, I think we should discuss how users will provide backend options for the multiple backends that are created. I think we may need a dict mapping device->option for the backends that are created?

Another question to resolve is how will this work for current users that are passing in the existing process group options. What is the transition plan for that and how can we provide sensible defaults?

Reply:
Well this PR is just for renaming. How should we continue this discussion down the road? Also I remember @shuqiangzhang is asking if we really have use cases when multiple backends that are created.

Reply:
I think let's discuss an ideal behavior first and then we can see what transition plan we can have here.